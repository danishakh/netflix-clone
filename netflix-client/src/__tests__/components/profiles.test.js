import React from 'react';
import { render } from '@testing-library/react'
import { Profiles } from '../../components';

describe('<Profiles />', () => {
    it('renders the <Profiles /> with populated data', () => {
        const { container, getByTestId, getByText } = render(
            <Profiles>
                <Profiles.Title>Who's watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.Item onClick={() => setProfile({ displayName: user.displayName, photoURL: user.photoURL})}>
                        <Profiles.Picture src='/images/karl.png' data-testid='profile-picture'/>
                        <Profiles.Name>John Cena</Profiles.Name>
                    </Profiles.Item>
                </Profiles.List>
            </Profiles>
        );
        
        expect(getByText("Who's watching?"));
        expect(getByTestId('profile-picture')).toBeTruthy();
        expect(getByText('John Cena')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <Profiles /> with populated data but misc profile image', () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who's watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.Item onClick={() => {}}>
                        <Profiles.Picture data-testid='profile-picture-misc'/>
                        <Profiles.Name>John Cena</Profiles.Name>
                    </Profiles.Item>
                </Profiles.List>
            </Profiles>
        );
        
        expect(getByText("Who's watching?"));
        expect(getByTestId('profile-picture-misc')).toBeTruthy();
        expect(getByText('John Cena')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();

    })
})
