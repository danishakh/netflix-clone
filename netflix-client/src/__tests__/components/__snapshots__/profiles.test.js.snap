// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-sc-1wnd71h-0 jTnGic"
>
  <h1
    class="profiles__Title-sc-1wnd71h-1 jCRVKF"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-1wnd71h-2 fBuFJw"
  >
    <li
      class="profiles__Item-sc-1wnd71h-5 jFdEtq"
    >
      <img
        class="profiles__Picture-sc-1wnd71h-4 kZeYlL"
        data-testid="profile-picture"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="profiles__Name-sc-1wnd71h-3 djxzCl"
      >
        John Cena
      </p>
    </li>
  </ul>
</div>
`;
